org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -190,6 +194,30 @@ public Charset getCharset() {
     public <G> G getClientObject() {
       return (G) file;
     }
+
+    @Override
+    public String contents() throws IOException {
+      try (TextFileContext context = new TextFileContext(file)) {
+        return context.getDocument().get();
+      } catch (CoreException e) {
+        // ignore
+      }
+      return null;
```

Comment:
According to the implemented interface, it shouldn't return null (should we change it?)